cms/djangoapps/contentstore/rest_api/v2/views/home.py

```diff
@@ -0,0 +1,147 @@
+    """Custom paginator for the home page courses view version 2."""
+
+    def get_paginated_response(self, data):
+        """Return a paginated style `Response` object for the given output data."""
+        return Response(OrderedDict([
+            ('count', self.page.paginator.count),
+            ('num_pages', self.page.paginator.num_pages),
+            ('next', self.get_next_link()),
+            ('previous', self.get_previous_link()),
+            ('results', data),
+        ]))
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """
+        Paginate a queryset if required, either returning a page object,
+        or `None` if pagination is not configured for this view.
+
+        This method is a modified version of the original `paginate_queryset` method
+        from the `PageNumberPagination` class. The original method was modified to
+        handle the case where the `queryset` is a `filter` object.
+        """
+        if isinstance(queryset, filter):
+            queryset = list(queryset)
+
+        return super().paginate_queryset(queryset, request, view)
+
+
+@view_auth_classes(is_authenticated=True)
+class HomePageCoursesViewV2(APIView):
+    """View for getting all courses available to the logged in user."""
+
+    @apidocs.schema(
+        parameters=[
+            apidocs.string_parameter(
+                "org",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to filter by course org",
+            ),
+            apidocs.string_parameter(
+                "search",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to filter by course name, org, or number",
+            ),
+            apidocs.string_parameter(
+                "order",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to order by course name, org, or number",
+            ),
+            apidocs.string_parameter(
+                "active_only",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to filter by active courses only",
+            ),
+            apidocs.string_parameter(
+                "archived_only",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to filter by archived courses only",
+            ),
+            apidocs.string_parameter(
+                "page",
+                apidocs.ParameterLocation.QUERY,
+                description="Query param to paginate the courses",
+            ),
+        ],
+        responses={
+            200: CourseHomeTabSerializerV2,
+            401: "The requester is not authenticated.",
+        },
+    )
+    def get(self, request: Request):
+        """
+        Get an object containing all courses.
+
+        **Example Request**
+
+            GET /api/contentstore/v2/home/courses
+            GET /api/contentstore/v2/home/courses?org=edX
```

Comment:
I tested all this query params but "org=text" does nothing. We should remove this line

Reply:
Below I found the `enable_global_staff_optimization` flag and tested again when the flag is set. This line queryparam does work and no issue here.